Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+        }
+        .overlay(alignment: .top) {
+            accuracyView
+                .multilineTextAlignment(.center)
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(8)
+                .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+    }
+    
+    /// If necessary, updates the scene view's camera controller for a new location coming
+    /// from the location datasource.
+    @MainActor
+    private func updateSceneView(for location: Location, heading: Double) {
+        // Do not use cached location more than 10 seconds old.
+        guard abs(lastLocationTimestamp?.timeIntervalSinceNow ?? 0) < 10 else { return }
+        
+        // Make sure that horizontal and vertical accuracy are valid.
+        guard location.horizontalAccuracy > validAccuracyThreshold,
+              location.verticalAccuracy > validAccuracyThreshold else { return }
+        
+        // Make sure either the initial camera is not set, or we need to update the camera.
+        guard !initialCameraIsSet || shouldUpdateCamera(for: location) else { return }
+        
+        // Add some of the vertical accuracy to the z value of the position, that way if the
+        // GPS location is not accurate, we won't end up below the earth's surface.
+        let altitude = (location.position.z ?? 0) + location.verticalAccuracy
+        
+        cameraController.originCamera = Camera(
+            latitude: location.position.y,
+            longitude: location.position.x,
+            altitude: altitude,
+            heading: calibrationHeading ?? heading,
+            pitch: 90,
+            roll: 0
+        )
+        
+        // We have to do this or the error gets bigger and bigger.
+        cameraController.transformationMatrix = .identity
+        arViewProxy.session.run(configuration, options: .resetTracking)
+        
+        // If initial camera is not set, then we set it the flag here to true
+        // and set the status text to empty.
+        if !initialCameraIsSet {
+            coachingOverlayIsActive = false
+            withAnimation {
+                initialCameraIsSet = true
+            }
+        }
+    }
+    
+    /// Returns a Boolean value indicating if the camera should be updated for a location
+    /// coming in from the location datasource.
+    func shouldUpdateCamera(for location: Location) -> Bool {
+        // Do not update unless the horizontal accuracy is less than a threshold.
+        guard let currentCamera,
+              let spatialReference = currentCamera.location.spatialReference,
+              let currentPosition = GeometryEngine.project(location.position, into: spatialReference)
+        else { return false }
+        
+        // Measure the distance between the location datasource's reported location
+        // and the camera's current location.
+        guard let result = GeometryEngine.geodeticDistance(
```

Comment:
I'm a bit wary of the geodetic calculation's performance. The `updateSceneView` method is called frequently. The most time consuming part in it is the construction of a new `Camera` object. While the time consumers in `shouldUpdateCamera` are the 2 `GeometryEngine` class methods.

We can probably run a performance test for both methods in release config, and see which takes longer. 🤔 Just a thought, I guess the time difference here is negligible compared to the frame rendering.

Reply:
Good point. The logic to update the heading can be removed from `updateSceneView` so that the method is called less frequently since the geodesic calculation does not use the heading value.